src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -276,6 +276,10 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
             catch (Exception ex)
             {
                 _stream.AbortWrite((long)Http3ErrorCode.InternalError);
+                if (ex is HttpRequestException)
```

Comment:
Code like this always makes me slightly queasy... feels like we are mixing exception usage across different layers of behavior. Not a huge deal, though.